### ISSUE TITLE
JSON encoding/decoding not consistent

### ISSUE THREAD

**Issue by mlenkeit** [eu-digital-green-certificates/dgc-certlogic-ios#13]
## Describe the bug

The encoding and decoding of a `Rule` is not consistent, i.e. not all fields are encoded, e.g.:

- `logic` is decoded but not encoded in `encode`
- `countryCode` is decoded not encoded in `encode`
- `affectedString` is encoded twice

https://github.com/eu-digital-green-certificates/dgc-certlogic-ios/blob/a4b10d4d591ae32989587836a3d002659a71c024/Sources/CertLogic/Rule.swift#L154-L185

## Expected behaviour

Encoding and decoding read/set the same fields/attributes.

## Steps to reproduce the issue

See code sample

## Technical details

n/a

## Possible Fix

use same fields for encoding and decoding

## Additional context

n/a

**Comment by ArturFriesen** [comment]
Unit test to reproduce the encoding issue:

```
    func testCodable() throws {
        let description = Description(lang: "", desc: "")
        let rule = Rule(identifier: "", type: "", version: "", schemaVersion: "", engine: "", engineVersion: "", certificateType: "", description: [description], validFrom: "", validTo: "", affectedString: [""], logic: JSON(""), countryCode: "")

        let encodedRule = try JSONEncoder().encode(rule)

        print("encodedRule: \(encodedRule)")

        let decodedRule = try JSONDecoder().decode(Rule.self, from: encodedRule)

        print("decodedRule \(decodedRule)")
    }
```

**Comment by alexchornyi** [comment]
fixed in #14

**Comment by mlenkeit** [comment]
@ArturFriesen  confirmed that it's fixed with #14, thanks @alexchornyi!
